src/modules/hip/kernel/warp_perspective.hpp

```diff
@@ -0,0 +1,482 @@
+}
+
+__device__ void warp_perspective_srclocs_hip_compute(float perspectiveMatrixElement, float4 locParComponent_f4, float4 roiComponent_f4, d_float8 locParC_f8, d_float8 *locSrcPtr_f8)
+{
+    d_float8 increment_f8;
+    increment_f8.f4[0] = make_float4(0, perspectiveMatrixElement, perspectiveMatrixElement + perspectiveMatrixElement, perspectiveMatrixElement + perspectiveMatrixElement + perspectiveMatrixElement);
+    increment_f8.f4[1] = (float4)(perspectiveMatrixElement + increment_f8.f4[0].w) + increment_f8.f4[0];
+    locSrcPtr_f8->f4[0] = ((locParComponent_f4 + increment_f8.f4[0])/locParC_f8.f4[0]) + roiComponent_f4;
+    locSrcPtr_f8->f4[1] = ((locParComponent_f4 + increment_f8.f4[1])/locParC_f8.f4[1]) + roiComponent_f4;
+}
+
+__device__ void warp_perspective_roi_and_srclocs_hip_compute(int4 *srcRoiPtr_i4, int id_x, int id_y, d_float9 *perspectiveMatrix_f9, d_float16 *locSrc_f16)
+{
+    float2 locDst_f2;
+    float3 locPar_f3;
+    d_float8 locParC_f8;
+    float roiHalfWidth = (srcRoiPtr_i4->z - srcRoiPtr_i4->x + 1) >> 1;
+    float roiHalfHeight = (srcRoiPtr_i4->w - srcRoiPtr_i4->y + 1) >> 1;
+    locDst_f2.x = (float) (id_x - roiHalfWidth);
+    locDst_f2.y = (float) (id_y - roiHalfHeight);
+    locPar_f3.x = fmaf(locDst_f2.x, perspectiveMatrix_f9->f1[0], fmaf(locDst_f2.y, perspectiveMatrix_f9->f1[1], perspectiveMatrix_f9->f1[2]));
+    locPar_f3.y = fmaf(locDst_f2.x, perspectiveMatrix_f9->f1[3], fmaf(locDst_f2.y, perspectiveMatrix_f9->f1[4], perspectiveMatrix_f9->f1[5]));
+    locPar_f3.z = fmaf(locDst_f2.x, perspectiveMatrix_f9->f1[6], fmaf(locDst_f2.y, perspectiveMatrix_f9->f1[7], perspectiveMatrix_f9->f1[8]));
+    warp_perspective_parcommon_hip_compute(perspectiveMatrix_f9->f1[6], (float4)locPar_f3.z, &locParC_f8);
+    warp_perspective_srclocs_hip_compute(perspectiveMatrix_f9->f1[0], (float4)locPar_f3.x, (float4)roiHalfWidth, locParC_f8, &(locSrc_f16->f8[0]));    // Compute 8 locSrcX
+    warp_perspective_srclocs_hip_compute(perspectiveMatrix_f9->f1[3], (float4)locPar_f3.y, (float4)roiHalfHeight, locParC_f8, &(locSrc_f16->f8[1]));    // Compute 8 locSrcY
+}
+
+// -------------------- Set 1 - Bilinear Interpolation --------------------
+
+template <typename T>
+__global__ void warp_perspective_bilinear_pkd_hip_tensor(T *srcPtr,
+                                                uint2 srcStridesNH,
+                                                T *dstPtr,
+                                                uint2 dstStridesNH,
+                                                uint2 dstDimsWH,
+                                                d_float9 *perspectiveTensor,
+                                                RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= dstDimsWH.y) || (id_x >= dstDimsWH.x))
```

Comment:
Lets validate with roiWidth and roiHeight itself, intead of validating with maxWidth and maxHeight
Applicable to all such instances in the PR
```suggestion
   if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
```

Reply:
Updated